Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -158,8 +171,11 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
 
     if (responseStatusCode != -1) {
         NSNumber *statusCode = [NSNumber numberWithInteger:responseStatusCode];
-        [netSpan setTagValue:[NSString stringWithFormat:@"%@", statusCode]
-                      forKey:@"http.status_code"];
+
+        if (netSpan != nil) {
+            [netSpan setTagValue:[NSString stringWithFormat:@"%@", statusCode]
+                          forKey:@"http.status_code"];
+        }
```

Comment:
h: Instead of checking for netSpan here, we should exit the function in an early moment if the task does not contains a netSpan.

I did some tests here, when the task is suspended or cancelled, the state is changed twice, once for suspended/cancelled and again for completed. We may create two breadcrumbs for each of this requests, I don't believe this is the desired behavior.

Reply:
I added a test `testWhenTaskCancelledAndSuspended_OnlyOneBreadcrumb`, which tests the behavior you described, and it only adds one breadcrumb. Can you maybe change the test to reproduce the issue you describe? Maybe it's not a problem anymore because we remove the observer earlier now.

Reply:
The same way is hard to reproduce the KVO error, would be difficult to reproduce two breadcrumbs errors, I cant reproduce two KVO calls in parallel, the same issue that would case the previous crash.